### PR TITLE
Fixed netfx System.Data.Common.Tests fails on non English Windows

### PR DESCRIPTION
**Pull request by AlexRadch**
Fixed netfx System.Data.Common.Tests fails on non English Windows. See https://github.com/dotnet/corefx/issues/28136.

### CODE REVIEW

**Comment by AlexRadch** [comment]
@dotnet-bot test Linux x64 Release Build 
@dotnet-bot test OSX x64 Debug Build

**Comment by AlexRadch** [comment]
@dotnet-bot test Linux x64 Release Build

File: src/System.Data.Common/tests/System/Data/Common/DbConnectionStringBuilderTest.cs [hunk 1]
```diff
@@ -319,7 +320,7 @@ public void Indexer_Keyword_Invalid()
                     Assert.Equal(typeof(ArgumentException), ex.GetType());
                     Assert.Null(ex.InnerException);
                     Assert.NotNull(ex.Message);
-                    Assert.True(ex.Message.IndexOf("'" + keyword + "'") != -1);
+                    Assert.True(Regex.IsMatch(ex.Message, @"[\p{Pi}\p{Po}]" + Regex.Escape(keyword) + @"[\p{Pf}\p{Po}]"));
```
Comment (saurabh500) [inline src/System.Data.Common/tests/System/Data/Common/DbConnectionStringBuilderTest.cs]: Can you explain this regex and how it fixes the issue in the PR description?

**Comment by AlexRadch** [comment]
@dotnet-bot test Windows x64 Debug Build

**Comment by AlexRadch** [comment]
@dotnet-bot test Windows x86 Release Build

**Comment by AlexRadch** [comment]
Can it be committed?

**Pull request merged**